rbd/rbd.go

```diff
@@ -799,7 +800,11 @@ func (image *Image) Seek(offset int64, whence int) (int64, error) {
 	return image.offset, nil
 }
 
-// int rbd_discard(rbd_image_t image, uint64_t ofs, uint64_t len);
+// Discard the supplied range from the image. The result is similar to filling
+// the range with zeroes.
```

Comment:
"filling the range with zeros" is quite different. "discard" punches holes so that the backing storage can be released, filling zeros does not have the "free space" kind of effect. It is true that holes created with Discard are read back as zeros though.

Suggestion: Discard the supplied range of the image so that the backing storage can free-up space. The supplied range will be read as zeros once Discard returns.

Reply:
Heh. I adapted the text from the python ceph bindings which has, "Trim the range from the image. It will be logically filled with zeroes."
Your suggestion is good though. I will take most of it.